modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadLaws.kt

```diff
@@ -72,9 +81,25 @@ object MonadLaws {
       }.equalUnderTheLaw(just(num + 2), EQ)
     }
 
-  fun <F> Monad<F>.selectEQSelectM(EQ: Eq<Kind<F, Int>>): Unit =
+  fun <F> Monad<F>.preservesSelective(SL: Selective<F>, EQ: Eq<Kind<F, Int>>): Unit =
     forAll(Gen.either(Gen.int(), Gen.int())) { either ->
       val f = just<(Int) -> Int>(::identity)
-      just(either).select(f).equalUnderTheLaw(just(either).selectM(f), EQ)
+      SL.run { just(either).select(f) }.equalUnderTheLaw(just(either).select(f), EQ)
+    }
+
+  fun <F> Monad<F>.preservesFunctor(FF: Functor<F>, EQ: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.int(), Gen.functionAToB<Int, Int>(Gen.int())) { num, f ->
+      FF.run { just(num).map(f) }.equalUnderTheLaw(just(num).map(f), EQ)
+    }
+
+  fun <F> Monad<F>.preservesApplicative(AP: Applicative<F>, EQ: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.int(), Gen.functionAToB<Int, Int>(Gen.int())) { num, f ->
+      val ff = just(f)
+      AP.run { just(num).ap(ff) }.equalUnderTheLaw(just(num).ap(ff), EQ)
+
+      val fg = just(1)
+      AP.run { just(num).followedBy(fg) }.equalUnderTheLaw(just(num).followedBy(fg), EQ)
+
+      AP.run { just(num).apTap(fg) }.equalUnderTheLaw(just(num).apTap(fg), EQ)
```

Comment:
Can you name this "preservesApply" and change the constraint? We can be less strict about this in docs because most datatypes are both and Applicative is just more common, but in the api we should always use the lowest possible constraint. With that I think this is good to go.

Reply:
Btw this will result in false-positives until #1844 comes along and changes the way test-data is generated. Why? Because this currently only tests cases equal to `Id(num).applyAp(constFn) == Id(num).monadAp(constFn)`. You can ignore this for now as it's quite the large change to properly do away with that issue, but please change the monad version of ap to `flatMap { a -> ff.map { f -> f(a) } }` instead of the reverse order.